### ISSUE TITLE
Support -D_FORTIFY_SOURCE=3

### ISSUE THREAD

**Issue by marxin** [opensvc/multipath-tools#29]
I'm experimenting with `-D_FORTIFY_SOURCE=3` that will be available since GCC `12.1` and I see the following error:

```
[   18s] cc --std=gnu99  -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type  -g -Werror -Wall -Wextra -Wformat=2 -Wformat-overflow=2 -Werror=implicit-int -Werror=implicit-function-declaration -Werror=format-security -Wno-clobbered -Wno-error=clobbered -Werror=cast-qual -Werror=discarded-qualifiers -pipe -DBIN_DIR=\"/usr/sbin\" -DLIB_STRING=\"lib64\" -DRUN_DIR=\"run\" -MMD -MP -fPIE -DPIE -I. -I../libmultipath -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -DLIBDM_API_COOKIE -Wp,-D_FORTIFY_SOURCE=2 -c -o unixware.o unixware.c
[   18s] <command-line>: error: "_FORTIFY_SOURCE" redefined [-Werror]
[   18s] <command-line>: note: this is the location of the previous definition
```

Note I pass the `-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3` as part of `make OPTFLAGS=...`

@siddhesh

**Comment by mwilck** [comment]
Well, it's not exactly an "OPTFLAG", right? Perhaps you need to patch `Makefile.inc`.

**Comment by marxin** [comment]
I think the simplest patch would be changing `-Wp,-D_FORTIFY_SOURCE=2` to `-D_FORTIFY_SOURCE=2`. Then one would be able to unset it with `U_FORTIFY_SOURCE`.

**Comment by mwilck** [comment]
Sounds ok. Would you send a patch to dm-devel (cc me & bmarzins@redhat.com)?

**Comment by mwilck** [comment]
@bmarzins , FTR: the `-Wp` syntax has been added by your commit 1fce669 ("multipath: Build with standard rpm cflags") a long time ago. Any idea why you were using this rather than simply `-D_FORTIFY_SOURCE=2` ?

**Comment by bmarzins** [comment]
Don't recall, and I have no objection to changing it.

**Comment by marxin** [comment]
I can confirm that `gcc -Wp,-D_FORTIFY_SOURCE=2 -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 a.c -Werror` works, but still, I would like to have a unified approach for enabling of `-D_FORTIFY_SOURCE=3`. So please accept my patch.

**Comment by mwilck** [comment]
pushed to [openSUSE/multipath-tools (queue)](https://github.com/openSUSE/multipath-tools/tree/queue). Will be included in next bunch of fixes.

**Comment by mwilck** [comment]
It turns out that the proposed patch didn't work as intended. Updated the fix for #29 to d13d154.
